sdk/src/main/java/com/hedera/hashgraph/sdk/Executable.java

```diff
@@ -721,11 +729,15 @@ class GrpcRequest {
         public CallOptions getCallOptions() {
             var options = CallOptions.DEFAULT;
 
-            if (Executable.this.grpcDeadline != null) {
-                return options.withDeadlineAfter(Executable.this.grpcDeadline.toMillis(), TimeUnit.MILLISECONDS);
-            } else {
+            if (this.grpcDeadline == null && Executable.this.grpcDeadline == null ) {
                 return options;
             }
+
+            long deadline = Math.min(
+                this.grpcDeadline == null ? Long.MAX_VALUE : this.grpcDeadline.toMillis(),
+                Executable.this.grpcDeadline == null ? Long.MAX_VALUE : Executable.this.grpcDeadline.toMillis());
```

Comment:
How can either of these be null? Client has a default and we don't allow null. Executable inherits from client if not set and we don't allow null. GrpcRequest inherits from Executable. Should remove all nullable logic in GrpcRequest.

Reply:
GrpcRequest constructor currently allows null and the null check logic was already in place so I kept it. But yes, we can ensure no null values are permitted and clean-up the logic.